store/storage_manager_builder.go

```diff
@@ -0,0 +1,373 @@
+	}
+
+	return ethClient, nil
+}
+
+func (smb *StorageManagerBuilder) buildRelayPayloadRetriever(
+	ethClient common_eigenda.EthClient,
+	g1Srs []bn254.G1Affine,
+) (*clients_v2.RelayPayloadRetriever, error) {
+	relayClient, err := smb.buildRelayClient(ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("build relay client: %w", err)
+	}
+
+	relayPayloadRetriever, err := clients_v2.NewRelayPayloadRetriever(
+		smb.log,
+		//nolint:gosec // disable G404: this doesn't need to be cryptographically secure
+		rand.New(rand.NewSource(time.Now().UnixNano())),
+		smb.v2ClientCfg.RelayPayloadRetrieverCfg,
+		relayClient,
+		g1Srs)
+	if err != nil {
+		return nil, fmt.Errorf("new relay payload retriever: %w", err)
+	}
+
+	return relayPayloadRetriever, nil
+}
+
+func (smb *StorageManagerBuilder) buildRelayClient(ethClient common_eigenda.EthClient) (clients_v2.RelayClient, error) {
+	reader, err := eigenda_eth.NewReader(smb.log, ethClient, "0x0", smb.v2ClientCfg.ServiceManagerAddress)
+	if err != nil {
+		return nil, fmt.Errorf("new eth reader: %w", err)
+	}
+
+	relayURLProvider, err := relay.NewRelayUrlProvider(ethClient, reader.GetRelayRegistryAddress())
+	if err != nil {
+		return nil, fmt.Errorf("new relay url provider: %w", err)
+	}
+
+	relayCfg := &clients_v2.RelayClientConfig{
+		UseSecureGrpcFlag: smb.v2ClientCfg.DisperserClientCfg.UseSecureGrpcFlag,
+		// we should never expect a message greater than our allowed max blob size.
+		// 10% of max blob size is added for additional safety
+		MaxGRPCMessageSize: smb.maxBlobSizeBytes + (smb.maxBlobSizeBytes / 10),
+	}
+
+	relayClient, err := clients_v2.NewRelayClient(relayCfg, smb.log, relayURLProvider)
+	if err != nil {
+		return nil, fmt.Errorf("new relay client: %w", err)
+	}
+
+	return relayClient, nil
+}
+
+func (smb *StorageManagerBuilder) buildPayloadDisperser(
+	ctx context.Context,
+	ethClient common_eigenda.EthClient,
+	kzgProver *prover.Prover,
+	certVerifier *verification.CertVerifier,
+) (*clients_v2.PayloadDisperser, error) {
+	signer, err := smb.buildLocalSigner(ctx, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("build local signer: %w", err)
+	}
+
+	disperserClient, err := clients_v2.NewDisperserClient(&smb.v2ClientCfg.DisperserClientCfg, signer, kzgProver, nil)
+	if err != nil {
+		return nil, fmt.Errorf("new disperser client: %w", err)
+	}
+
+	payloadDisperser, err := clients_v2.NewPayloadDisperser(
+		smb.log,
+		smb.v2ClientCfg.PayloadDisperserCfg,
+		disperserClient,
+		certVerifier)
+	if err != nil {
+		return nil, fmt.Errorf("new payload disperser: %w", err)
+	}
+
+	return payloadDisperser, nil
+}
+
+// buildLocalSigner attempts to check the pending balance of the created signer account. If the check fails, or if the
+// balance is determined to be 0, the user is warned with a log. This method doesn't return an error based on this check:
+// it's possible that a user could want to set up a signer before it's actually ready to be used
```

Comment:
Can we add a comment explaining that this is a heuristic with a todo describing how we could extend this in the future to do checks against the payment vault or verify payment states through some accessor on the disperser's accountant

Reply:
[7384ac1](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/7384ac13b72ea64626f34dc04d71e3ff526fe790)